src/components/IstioWizards/MatchingRouting.tsx

```diff
@@ -0,0 +1,511 @@
+          onSelect={this.onSelectRoute}
+        >
+          {routeItems}
+        </DropdownButton>
+      </Form>
+    );
+  };
+
+  renderMatches = () => {
+    const matches: any[] = this.state.matches.map((match, index) => (
+      <span key={match + '-' + index}>
+        <Label className={labelMatchStyle} type="primary" onRemoveClick={() => this.onRemoveMatch(match)}>
+          {match}
+        </Label>{' '}
+      </span>
+    ));
+    return <div className={labelContainerStyle}>{matches}</div>;
+  };
+
+  renderRules = () => {
+    let ruleItems: any[] = [];
+    let isValid: boolean = true;
+    let matchAll: number = this.matchAllIndex(this.state.rules);
+    for (let index = 0; index < this.state.rules.length; index++) {
+      const rule = this.state.rules[index];
+      isValid = matchAll === -1 || index <= matchAll;
+      const matches: any[] = rule.matches.map((map, index) => {
+        return <small key={'match-' + map + '-' + index}>{map}</small>;
+      });
+      const ruleActions = (
+        <div>
+          <Button onClick={() => this.onRemoveRule(index)}>Remove</Button>
+          {this.state.rules.length > 1 && (
+            <DropdownKebab key={'move-rule-actions-' + index} id={'move-rule-actions-' + index} pullRight={true}>
+              {index > 0 && <MenuItem onClick={() => this.onMoveRule(index, MOVE_TYPE.UP)}>Move Up</MenuItem>}
+              {index + 1 < this.state.rules.length && (
+                <MenuItem onClick={() => this.onMoveRule(index, MOVE_TYPE.DOWN)}>Move Down</MenuItem>
+              )}
+            </DropdownKebab>
+          )}
+        </div>
+      );
+      ruleItems.push(
+        <ListViewItem
+          key={'match-rule-' + index}
+          leftContent={<ListViewIcon type={vsIconType} name={vsIconName} />}
+          heading={
+            <div>
+              Matches:
+              {rule.matches.length === 0 && <small>Any request</small>}
+              {rule.matches.length !== 0 && matches}
+            </div>
+          }
+          description={
+            <div>
+              <b>Route to:</b>
+              <div>
+                <span>
+                  <ListViewIcon
+                    type={rule.routeType === ROUTE_TYPE.SERVICE ? svcIconType : wkIconType}
+                    name={rule.routeType === ROUTE_TYPE.SERVICE ? svcIconName : wkIconName}
+                  />
+                  <span className={routeToStyle}>{rule.route}</span>
+                </span>
+              </div>
+              {!isValid && (
+                <div className={validationStyle}>
+                  <Icon type={valIconType} name={valIconName} /> Match 'Any request' is defined in a previous rule. This
+                  rule is not accessible.
+                </div>
+              )}
+            </div>
+          }
+          actions={ruleActions}
+        />
+      );
+    }
+    return (
+      <div>
+        <ListView>{ruleItems}</ListView>
+      </div>
+    );
+  };
+
+  render() {
```

Comment:
This component is pretty huge. Do you think it should be splitted in different ones? Break it down in smaller pieces?

Reply:
Probably in a next interation.
At the moment I want to maintain a first version with all state together, and then split it in minor pieces.